vmsyscall/src/bootinfo.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! The `bootinfo` module includes the hard-coded memory layout and
+//! the `BootInfo` structure which is handed to the kernel.
+//!
+//! Initially copied from:
+//! https://github.com/rust-osdev/bootloader/blob/90f5b8910d146d6d489b70a6341d778253663cfa/src/bootinfo/mod.rs
+
+use crate::memory_map::Map;
+use core::fmt;
+
+/// This structure represents the information that the bootloader passes to the kernel.
+///
+/// The information is passed as an argument to the entry point:
+///
+/// ```ignore
+/// pub extern "C" fn _start(boot_info: &'static BootInfo) -> ! {
+///    // […]
+/// }
+/// ```
+///
+/// Note that no type checking occurs for the entry point function, so be careful to
+/// use the correct argument types. To ensure that the entry point function has the correct
+/// signature, use the [`entry_point`] macro.
+#[derive(Clone)]
+#[repr(C)]
+pub struct BootInfo {
```

Comment:
This is just a note on general strategy. Where possible, we want the guest to determine layout.  So rather than the host doing the whole layout and passing it to the guest, the host should do minimal layout and allow the guest to determine things like the code location. This gives the host less advantage on attacking the guest.

The general principle is that the host should be in charge of allocation, but the guest should be in charge of its own layout.

Reply:
That is already the case. I reduced it to the bare minimum already.

Reply:
But yeah, I think `BootInfo` can be reduced to just `&ElfHeader` of the code.

Reply:
Ah, no... I need at least the reserved virtual memory range.

Reply:
The rest is nice to have, so I don't have to calculate it myself... like the range of the `shim` and the range of the `code`.